pkg/config/serving/server.go

```diff
@@ -20,23 +25,46 @@ func ToServerConfig(servingInfo configv1.HTTPServingInfo, authenticationConfig o
 	if err != nil {
 		return nil, err
 	}
+
 	if err := servingOptions.ApplyTo(&config.SecureServing, &config.LoopbackClientConfig); err != nil {
 		return nil, err
 	}
 
+	var lastApplyErr error
+
 	if !authenticationConfig.Disabled {
 		authenticationOptions := genericapiserveroptions.NewDelegatingAuthenticationOptions()
 		authenticationOptions.RemoteKubeConfigFile = kubeConfigFile
-		if err := authenticationOptions.ApplyTo(&config.Authentication, config.SecureServing, config.OpenAPIConfig); err != nil {
-			return nil, err
+
+		// In some cases the API server can return connection refused when getting the "extension-apiserver-authentication"
+		// config map.
+		err := wait.PollImmediateUntil(20*time.Second, func() (done bool, err error) {
```

Comment:
don't wait forever.

Reply:
i don't AFAIK, there is 20s timeout on this + stopCh

Reply:
yeah, the stop channel won't close.

Reply:
We prefer crash looping to waiting forever.  PollImmediate, every one second for 15 seconds, then crashloop